docs/cisco.ios.ios_user_module.rst

```diff
@@ -548,38 +685,139 @@ Examples
+
+    # router-ios#show running-config | section username
+    # username admin privilege 15 password 0 password
+    # username netop privilege 15 password 0 newpassword
+
+    # Using state: present
+
+    # Before state:
+    # -------------
+
+    # router-ios#show running-config | section ^username
+    # username admin privilege 15 password 0 password
+    # username netop privilege 15 password 0 password
+    # username netend privilege 15 password 0 password
+
+    # Present state set user view/role for users play:
+    # --------------------------------------------
+
+    - name: Set user view/role for users
       cisco.ios.ios_user:
         aggregate:
-          - name: ansibletest2
-          - name: ansibletest3
+          - name: netop
+          - name: netend
         view: network-admin
+        state: present
 
-    - name: Add a user specifying password type
-      cisco.ios.ios_user:
-        name: ansibletest4
-        configured_password: "{{ new_password }}"
-        password_type: password
+    # Commands Fired:
+    # ---------------
 
-    - name: Add a user with MD5 hashed password
+    # "commands": [
+    #     "username netop view network-admin",
+    #     "username netend view network-admin"
+    # ],
+
+    # After state:
+    # ------------
+
+    # router-ios#show running-config | section username
+    # username admin privilege 15 password 0 password
+    # username netop privilege 15 view network-admin password 0 password
+    # username netend privilege 15 view network-admin password 0 password
+
+    # Using state: present
+
+    # Before state:
+    # -------------
+
+    # router-ios#show running-config | section ^username
+    # username admin privilege 15 password 0 password
+
+    # Present state create a new user with hashed password play:
+    # --------------------------------------------------------------
+
+    - name: Create a new user with hashed password
       cisco.ios.ios_user:
         name: ansibletest5
         hashed_password:
-          type: 5
-          value: $3$8JcDilcYgFZi.yz4ApaqkHG2.8/
+          type: 9
+          value: "$9$yuUxFM0JeIpU1U$mptJ5lbTLKWdEzohzY2SalIXlYbC1/gCiVHoLQax6NU"
+        state: present
+
+    # Commands Fired:
+    # ---------------
+
+    # "commands": [
+    #     "username ansibletest5 secret 9 $9$yuUxFM0JeIpU1U$mptJ5lbTLKWdEzohzY2SalIXlYbC1/gCiVHoLQax6NU"
+    # ],
+
+    # After state:
+    # ------------
+
+    # router-ios#show running-config | section username
+    # username admin privilege 15 password 0 password
+    # username ansibletest5 secret 9 $9$yuUxFM0JeIpU1U$mptJ5lbTLKWdEzohzY2SalIXlYbC1/gCiVHoLQax6NU
+
+    # Using state: absent
+
+    # Before state:
+    # -------------
+
+    # router-ios#show running-config | section ^username
+    # username admin privilege 15 password 0 password
+    # username ansibletest1 password 0 password
+    # username ansibletest2 secret 9 $9$yuUxFM0JeIpU1U$mptJ5lbTLKWdEzohzY2SalIXlYbC1/gCiVHoLQax6NU
```

Comment:
GH security leak would complain, make this a random string and not a real secret, to dodge that situation, it's alright if the secret is not real.